src/components/StoreIssueBlock.js

```diff
@@ -3,52 +3,66 @@ import PropTypes from "prop-types";
 import HashColor from "./UI/HashColor";
 import HashColorInput from "./UI/HashColorInput";
 import { OrangeButton } from "./UI/Button";
+import { isValidHash } from "../components/utils";
 
 class StoreIssueBlock extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      certificateHash: ""
+      certificateHash: "",
+      certificateHashIsValid: true
     };
 
     this.onHashChange = this.onHashChange.bind(this);
-    this.onIssueClick = this.onIssueClick.bind(this);
   }
 
   onHashChange(event) {
     this.setState({
-      certificateHash: event.target.value
-    });
-  }
-
-  onIssueClick() {
-    const { adminAddress, storeAddress, handleCertificateIssue } = this.props;
-    handleCertificateIssue({
-      storeAddress,
-      fromAddress: adminAddress,
-      certificateHash: this.state.certificateHash
+      certificateHash: event.target.value,
+      certificateHashIsValid: isValidHash(event.target.value)
     });
   }
 
   render() {
+    const certificateHashMessage = this.state.certificateHashIsValid
+      ? ""
+      : "Merkle Root Hash is not valid.";
+
+    const onIssueClick = () => {
+      const { adminAddress, storeAddress, handleCertificateIssue } = this.props;
+      const { certificateHash, certificateHashIsValid } = this.state;
+      if (isValidHash(certificateHash) && certificateHashIsValid) {
```

Comment:
Same concern as above because `certificateHashIsValid` is really just checking if state has changed.